# **Graded Challenge 5**

Nama  : Zafirah Aida Adista

Batch : CODA-RMT-015

Program ini dibuat untuk  mengevaluasi konsep Business Knowledge dan Practical Statistics

## **Case Study - Analisis Penjualan Produk Seblak di Tokopedia**
Rosie ingin menambah pendapatanmu dengan berjualan. Dengan modal terbatas yang hanya cukup untuk biaya promosi, ia memutuskan untuk menjalankan bisnis **dropship** di platform Tokopedia.

Melihat tren media sosial yang ramai membicarakan seblak, mulai dari video mukbang, review kuliner, hingga challenge makan pedas, Rosie tertarik untuk menjadikan produk **seblak** sebagai produk jualannya. Namun sebelum mengambil keputusan, ia ingin memastikan keputusan bisnisnya berdasarkan **data**, bukan sekadar asumsi viral di media sosial.

Karena Rosie lulusan bootcamp Data Analytics Hacktiv8, dengan kemampuan dan pengetahuannya, Rosie ingin menganalisis bagaimana penjualan produk seblak di Tokopedia. Ia ingin menjawab pertanyaan-pertanyaan krusial sebelum memulai berjualan.


## A. Business Understanding


### Analisis SMART Framework
*   **Specific:** Menganalisis karakteristik penjualan produk seblak di Tokopedia, mencakup distribusi harga, volume penjualan, rating, potensi pendapatan, perbedaan harga berdasarkan wilayah, dan hubungan antara harga dengan banyaknya produk terjual

*   **Measurable**: Menggunakan confidence interval 95%, perbedaan harga antar  wilayah dengan α = 0,05, dan koefisien korelasi Spearman

*   **Achievable**: Analisis dapat dilakukan menggunakan dataset hasil web scraping Tokopedia dengan metode statistik deskriptif, uji hipotesis, dan korelasi

*  **Relevant**: Hasil analisis dapat digunakan untuk menentukan apakah bisnis dropship seblak layak dijalankan, produk harga berapa yang sebaiknya dipilih, dan faktor yang bisa mempengaruhi penjualan seblak

*   **Time-bound**: Keputusan bisnis diambil dalam satu bulan ke depan berdasarkan data penjualan yang berjalan


### Problem Statement

Rosie akan menganalisis karakteristik pasar seblak di Tokopedia menggunakan metode statistik deskriptif, uji hipotesis, dan korelasi Spearman untuk menentukan kelayakan bisnis dropship, strategi penetapan harga, dan mengetahui faktor yang bisa mempengaruhi penjulan yang seluruh prosesnya akan diselesaikan untuk pengambilan keputusan bisnis dalam waktu satu bulan.

## B. Analisis Data


In [2]:
# Import Libraries
import numpy as np
import pandas as pd
from scipy import stats
import requests

In [3]:
# Load data
df = pd.read_csv('P0G3_Zafirah Aida_data bersih.csv')

### Analisis 1: Statistik Deskriptif (Mengenal Karakteristik Data)

Langkah pertama sebelum mengambil keputusan bisnis adalah memahami seperti apa data yang kita miliki. Rosie ingin mengetahui gambaran umum tiga variabel kunci: **harga produk**, **banyaknya produk terjual**, dan **rating produk**. Ia menghitung rata-rata, median, standar deviasi, skewness, dan kurtosis untuk memahami distribusi dan mendeteksi adanya outlier pada data.

In [4]:
# Membuat dictionary kolom yang akan dianalisis
kolom_analisis = {
    'harga produk': 'Harga Produk',
    'banyaknya terjual': 'Banyak Produk Terjual',
    'rating produk': 'Rating Produk'
}

# Membuat list kosong untuk menampung semua hasil perhitungan dari looping
hasil = []

# Looping untuk mencari Mean, Median,  Standar Deviasi, Skewness, dan Kurtosis pada masing-masing kolom
for kolom, label in kolom_analisis.items():
    data = df[kolom]
    hasil.append({
        'Variabel': label,
        'Mean': round(data.mean(), 2),
        'Median': round(data.median(), 2),
        'Std Dev': round(data.std(), 2),
        'Skewness': round(data.skew(), 4),
        'Kurtosis': round(data.kurt(), 4)
    })

# Mengubah List menjadi df
df_desc = pd.DataFrame(hasil).set_index('Variabel')

# Menampilkan df
df_desc

,Mean,Median,Std Dev,Skewness,Kurtosis
Variabel,,,,,
Harga Produk,19891.80,3740.0,26264.66,2.6086,10.0522
Banyak Produk Terjual,175.50,70.0,386.38,4.0025,16.2910
Rating Produk,4.94,5.0,0.09,-1.0962,-0.2933


**Insight Analisis 1**

**Harga Produk:**
- Rata-rata harga sebesar **Rp 19.891,80** namun median jauh lebih rendah di **Rp 3.740**. Kesenjangan besar antara mean dan median ini mengindikasikan data **condong ke kanan (right-skewed)**, terkonfirmasi dengan nilai skewness **+2.61**.
- Kurtosis sebesar **10.05** (leptokurtic) menunjukkan distribusi memiliki ekor yang sangat panjang, menandakan adanya **outlier di sisi harga tinggi**. Ini terjadi karena ada produk grosir seperti karton Indomie Seblak seharga Rp 153.000 di antara mayoritas produk murah Indomie satuan.
- Standar deviasi **Rp 26.264** yang jauh lebih besar dari mean memperkuat indikasi outlier tersebut.

**Banyak Produk Terjual:**
- Rata-rata terjual **175,5 unit/bulan** dengan median hanya **70 unit**. Skewness **+4.00** dan kurtosis **16.29** menunjukkan distribusi **sangat condong ke kanan** dengan outlier ekstrem. Ada beberapa produk yang terjual sangat banyak (> 1.000 unit/bulan) sehingga menarik mean ke atas.
- Outlier ini bersifat alami karena adanya produk yang dijual satuan.

**Rating Produk:**
- Rata-rata rating **4.94** mendekati nilai maksimum 5.0, dengan median **5.0** dan standar deviasi yang sangat kecil (**0.09**). Ini menunjukkan hampir semua produk seblak mendapat rating sangat tinggi.
- Skewness **-1.10** (negatif/left-skewed) menandakan sebagian besar nilai terkonsentrasi di sisi kanan (rating tinggi), sejalan dengan insight di atas.
- **Kesimpulan bisnis:** Konsumen secara umum **puas** dengan produk seblak yang dijual di Tokopedia. Ini menunjukkan bahwa produk seblak memang banyak diminati dan diterima baik oleh pasar.

### Analisis 2: Confidence Interval Pendapatan (Potensi pendapatan Dropship)

Setelah memahami karakteristik data, Rosie ingin menjawab pertanyaan yang paling krusial sebelum memulai bisnis: **"Berapa kisaran pendapatan bulanan yang bisa ia harapkan?"** Ia menghitung pendapatan per produk (harga x unit terjual) lalu menggunakan **Confidence Interval 95%** untuk mendapatkan estimasi rentang pendapatan dengan asumsi data terdistribusi normal dan data terjual merupakan penjualan per bulan.

In [5]:
# Menghitung pendapatan per produk
df['pendapatan'] = df['harga produk'] * df['banyaknya terjual']

# Menghitung standar deviasi
std_pendapatan = df['pendapatan'].std()
mean_pendapatan = df['pendapatan'].mean()

# Menghitung
n = len(df['pendapatan'])
low, up = stats.norm.interval(0.95,loc=mean_pendapatan,scale=std_pendapatan/np.sqrt(n))
print('Lower Limit:',low)
print('Upper Limit:',up)

Lower Limit: 1284355.873634201
Upper Limit: 7635361.459699133


Dengan tingkat kepercayaan **95%**, rata-rata pendapatan
populasi produk seblak per bulan berada pada rentang:

**Rp 1,284,356** s.d. **Rp 7,635,361**

**Insight Analisis 2**

Dengan **confidence interval 95%**, estimasi rata-rata pendapatan bulanan per produk seblak yang dijual di Tokopedia berada pada rentang **Rp 1.284.355,87 hingga Rp 7.635.361,46**.

Artinya:
- Dalam skenario **minimum (lower bound)**: dropshipper bisa mengharapkan pendapatan sekitar **Rp 1,28 juta/bulan** per produk yang dijual
- Dalam skenario **maksimum (upper bound)**: pendapatan bisa mencapai **Rp 7,63 juta/bulan** per produk
- Rentang yang lebar ini mencerminkan **variasi lebar** antar produk. Ada produk dengan harga satuan murah (Indomie Seblak ~Rp 3.500) dengan volume besar, dan ada produk grosir (Indomie karton isi 40 ~Rp 153.000) dengan harga tinggi namun volume lebih kecil.


### Analisis 3: Uji Hipotesis. Apakah Harga di Jabodetabek Berbeda dari Luar Jabodetabek?


Rosie tinggal di area Jabodetabek dan penasaran apakah terdapat perbedaan harga dengan penjual dari luar Jabodetabek. Biaya bahan baku, logistik, dan sewa di Jabodetabek umumnya lebih tinggi. Apakah ini mempengaruhi harga jual pada produk seblak?

**Hipotesis:**
- **H0:** Tidak ada perbedaan rata-rata harga produk seblak antara penjual di Jabodetabek dan di luar Jabodetabek
- **H1:** Terdapat perbedaan rata-rata harga produk seblak antara penjual di Jabodetabek dan di luar Jabodetabek

**Jenis uji:** Two-tailed, karena untuk mengetahui perbedaan antara wilayah Jabodetabek dan luar Jabodetabek (belum memiliki asumsi)

**Tingkat signifikansi:** α = 0.05

In [6]:
# Membuat daftar nama kota yang termasuk kategori Jabodetabek
kota_jabodetabek = [
    'Jakarta Selatan', 'Jakarta Timur', 'Jakarta Barat',
    'Jakarta Utara', 'Jakarta Pusat', 'Depok', 'Bekasi',
    'Tangerang', 'Tangerang Selatan', 'Bogor', 'Kab. Bekasi']

 # Membuat kolom baru 'wilayah' menggunakan fungsi lambda untuk memberi label pada data
df['wilayah'] = df['kota toko'].apply(lambda x: 'Jabodetabek' if x in kota_jabodetabek else 'Luar Jabodetabek')

# Memisahkan data harga ke dalam dua variabel berbeda untuk dibandingkan
jabod = df[df['wilayah'] == 'Jabodetabek']['harga produk']
luar  = df[df['wilayah'] == 'Luar Jabodetabek']['harga produk']

# Menghitung jumlah produk dan median pada dua variabel
print(f'Jumlah produk Jabodetabek: {len(jabod)} produk')
print(f'Median harga Jabodetabek: Rp {jabod.median():,.2f}') # Menggunakan median karena terdapat outlier
print(f'Jumlah produk Luar Jabodetabek: {len(luar)} produk')
print(f'Median harga Luar Jabodetabek: Rp {luar.median():,.2f}') # Menggunakan median karena terdapat outlier


Jumlah produk Jabodetabek: 13 produk
Median harga Jabodetabek: Rp 28,600.00
Jumlah produk Luar Jabodetabek: 47 produk
Median harga Luar Jabodetabek: Rp 3,740.00


In [13]:
# Uji Normalitas pada data menggunakan Shapiro-Wilk (agar bisa menentukan metode uji hipotesis yang digunakan)
stat_j, p_j = stats.shapiro(jabod)
stat_l, p_l = stats.shapiro(luar)
print(f'Jabodetabek : stat={stat_j:.4f}, p-value={p_j:.4f} → {"Normal" if p_j > 0.05 else "Tidak Normal"}')
print(f'Luar Jabod  : stat={stat_l:.4f}, p-value={p_l:.4f} → {"Normal" if p_l > 0.05 else "Tidak Normal"}')

Jabodetabek : stat=0.8083, p-value=0.0086 → Tidak Normal
Luar Jabod  : stat=0.6169, p-value=0.0000 → Tidak Normal


Kedua variabel menunjukkan distribusi yang **tidak normal**. Sehingga untuk uji hipotesis menggunakan metode **Mann-Whitney U Test** (pengganti metode T-test)

In [7]:
# Menjalankan uji Mann-Whitney U Test
stat_test, p_val = stats.mannwhitneyu(jabod, luar, alternative='two-sided')

# Menampilkan  hasil statistik
print(f'Statistic = {stat_test:.4f}')
print(f'p-value   = {p_val:.4f}')

Statistic = 306.5000
p-value   = 0.9928


**Kesimpulan**: Karena p-value (0.9928) > 0.05 maka menerima H0 → **Tidak ada perbedaan signifikan**

**Insight Analisis 3**

Rosie bebas memilih supplier dari wilayah mana saja tanpa perlu khawatir modalnya membengkak karena lokasi, karena **tidak ada perbedaan signifikan antara produk yang berasal dari Jabodetabek dan Luar Jabodetabek**. Ia bisa lebih fokus mencari supplier yang memiliki **pengiriman paling cepat** atau **biaya ongkir termurah** ke pelanggan daripada pusing memikirkan perbedaan harga produk antar wilayah.

### Analisis 4: Uji Korelasi. Apakah Konsumen Lebih Suka Produk Murah?


Setelah Rosie menyadari bahwa lokasi supplier tidak jadi masalah, ia kini menghadapi pertanyaan paling krusial bagi setiap pedagang: **"Seberapa sensitifkah pembeli seblak terhadap harga?"**
Rosie bimbang. Jika ia mengambil keuntungan (margin) terlalu besar, ia takut seblaknya tidak dilirik. Namun, jika terlalu murah, ia khawatir profitnya tidak cukup untuk biaya operasional (listrik, internet). Sebagai data analyst, Rosie tidak mau menebak-nebak. Ia pun melakukan **Uji Korelasi Spearman** untuk melihat hubungan antara harga dengan jumlah produk terjual.

**Hipotesis**:

**H0** : Harga produk tidak mempengaruhi banyaknya produk terjual

**H1** : Harga produk mempengaruhi banyaknya produk terjual

In [10]:
# Menjalankan Uji Korelasi Spearman
corr_s, p_s = stats.spearmanr(df['harga produk'], df['banyaknya terjual'])

print(f'Koefisien Korelasi (r) = {corr_s:.4f}')
print(f'p-value                = {p_s:.4f}')

Koefisien Korelasi (r) = -0.0029
p-value                = 0.9825


**Kesimpulan:**

Karena p-value (0.9825) > 0.05 maka:

*   Korelasi TIDAK SIGNIFIKAN → Tolak H0 dan menerima H1
*   **Tidak ada hubungan** linear yang bermakna antara harga dan rating



**Insight Analisis 4**

Karena p-value > 0.05, Rosie menyimpulkan bahwa **harga produk tidak memiliki pengaruh yang signifikan terhadap jumlah penjualan** seblak di Tokopedia. Artinya, pembeli seblak tidak semata-mata mencari harga yang paling murah.

Rosie menyadari bahwa ia tidak perlu terpaku pada strategi "jual yang paling murah untuk agar banyak yang beli" yang bisa merugikan margin keuntungannya. Daripada menjual dengan harga murah, Rosie memutuskan untuk menetapkan harga yang wajar (kompetitif namun tetap menguntungkan) dan akan lebih fokus pada faktor lain yang lebih diperhatikan konsumen, seperti kualitas foto produk, kecepatan respon chat, rating produk, karena tidak ada bukti statistik kalau harga murah bisa membuat produk menjadi laris.

## KESIMPULAN

Berdasarkan keempat analisis di atas, Rosie dapat menarik kesimpulan berikut sebelum memulai bisnis dropship seblak:

1. **Pasar Seblak Sangat Menjanjikan** : Rating rata-rata 4.94/5.0 menunjukkan kepuasan konsumen yang sangat tinggi. Produk seblak di Tokopedia sudah terbukti disukai pembelinya.

2. **Potensi Pendapatan Bulanan Rp 1,28 Juta - Rp 7,63 Juta** per produk (CI 95%). Dengan menjual beberapa SKU, Rosie berpotensi menghasilkan pendapatan yang signifikan.

3. **Tidak Ada Perbedaan Harga Berdasarkan Wilayah**:  Uji Mann-Whitney U Test menunjukkan tidak ada perbedaan harga yang signifikan antara Jabodetabek dan luar Jabodetabek, sehingga Rosie fleksibel dalam memilih supplier dari mana saja.

4. **Harga Bukan Faktor Penentu Banayknya Produk Terjual**:  Konsumen menilai kualitas, bukan harga semata. Rosie sebaiknya menjual produk berkualitas dengan margin yang baik, bukan sekadar yang paling murah.